src/main/java/com/cleanroommc/groovyscript/compat/inworldcrafting/jei/FluidRecipeCategory.java

```diff
@@ -0,0 +1,147 @@
+import com.cleanroommc.groovyscript.compat.inworldcrafting.FluidToFluid;
+import com.cleanroommc.groovyscript.compat.mods.jei.BaseCategory;
+import mezz.jei.api.IGuiHelper;
+import mezz.jei.api.gui.IDrawable;
+import mezz.jei.api.gui.IRecipeLayout;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.ingredients.VanillaTypes;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FluidRecipeCategory extends BaseCategory<FluidRecipeCategory.RecipeWrapper> {
+
+    public static final String UID = GroovyScript.ID + ":fluid_recipe";
+    public static final NumberFormat numberFormat = NumberFormat.getPercentInstance();
+
+    private static final int inputY = 23;
+    public static final int outputY = 52;
+    public static final int outputX = 105;
+
+    public final IDrawable downRightArrow;
+    public final IDrawable icon;
+
+    public FluidRecipeCategory(IGuiHelper guiHelper) {
+        super(guiHelper, UID, 176, 73);
+
+        downRightArrow = guiHelper.drawableBuilder(new ResourceLocation(GroovyScript.ID, "textures/jei/arrow_down_right.png"), 0, 0, 24, 18)
+                .setTextureSize(24, 18)
+                .build();
+        icon = guiHelper.createDrawableIngredient(new ItemStack(Items.WATER_BUCKET));
+    }
+
+    @Override
+    public void setRecipe(@NotNull IRecipeLayout recipeLayout, @NotNull RecipeWrapper recipeWrapper, @NotNull IIngredients ingredients) {
+        for (int i = 0; i < 9; i++) {
+            addItemSlot(recipeLayout, i, true, 7 + 18 * i, inputY);
+        }
+        addFluidSlot(recipeLayout, 10, true, 53, outputY);
+        if (recipeWrapper.recipe.getClass() == FluidToFluid.Recipe.class) {
+            addFluidSlot(recipeLayout, 11, false, outputX, outputY);
+        } else {
+            addItemSlot(recipeLayout, 11, false, outputX, outputY);
+        }
+        recipeLayout.getItemStacks().set(ingredients);
+        recipeLayout.getFluidStacks().set(ingredients);
+
+        setBackgrounds(recipeLayout.getItemStacks(), slot);
+        setBackgrounds(recipeLayout.getFluidStacks(), slot);
+
+        recipeLayout.getItemStacks().addTooltipCallback((slotIndex, input, ingredient, tooltip) -> {
+            if (slotIndex < 9) {
+                float chance = recipeWrapper.recipe.getItemConsumeChance()[slotIndex];
+                if (chance < 1) {
+                    tooltip.add(1, I18n.format("groovyscript.recipe.chance", numberFormat.format(chance)));
```

Comment:
`groovyscript.recipe.chance` doesnt exist in the lang file.
![image](https://github.com/CleanroomMC/GroovyScript/assets/25394029/4c53633e-3d9a-404e-9327-210712ab5393)